core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2336,6 +2335,11 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def maybeReturnOldClientError(error: Errors, requestVersion: Short): Boolean = {
+    requestVersion < 2 && (error == Errors.PRODUCER_FENCED || error == Errors
```

Comment:
We need to bump the protocol versions for the new error code. Otherwise older clients using version 2 will not know how to handle `TRANSACTION_TIMED_OUT` and will translate it to `UNKNOWN_SERVER_ERROR`, which is a regression.

Reply:
@abbccdda clarified offline that the version has already been bumped for 2.7, so this is not needed.